spotify_tensorflow/tfx/tfdv.py

```diff
@@ -71,7 +74,14 @@ def write_stats(self, pipeline_args):
                                                  stats_options=self.stats_options)
 
     def write_stats_and_schema(self, pipeline_args):  # type: (List[str]) -> None
-        self.write_stats(pipeline_args)
+        stats = self.write_stats(pipeline_args)
+        if not self.schema:
+            logger.warning(
+                "Inferring a new schema for this dataset. If you want to use an existing schema, "
+                "provide a value for schema_path in the constructor."
+            )
+            new_schema = tfdv.infer_schema(stats)
```

Comment:
do we want to add `infer_feature_shape=false` to be more conservative? but i don't think it's a big deal

Reply:
Good catch, fixed!